### PR TITLE
New module for FSx Lustre setup and PVC test

### PR DESCRIPTION
**Pull request by kornelbicskei**
*Issue #, if available:*

*Description of changes:*

Adding a new module to setup FSx Lustre CSI for EKS and installs a small app that tests PVC and S3 import/export

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: content/beginner/190_fsx_lustre/_index.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+weight: 190
+pre: '<i class="fa fa-film" aria-hidden="true"></i> '
+tags:
+  - beginner
+  - CON206
+---
+
+# Deploying Stateful Microservices with Amazon FSx Lustre
+
+{{< youtube ttCAqwnN_Qo >}}
```
Comment (brianhammons) [inline content/beginner/190_fsx_lustre/_index.md:13]: Can we leave this open for the video review?

File: content/beginner/190_fsx_lustre/launching-fsx.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+    *Take note of the policy Amazon Resource Name (ARN) that is returned.*
+
+3. Create a Kubernetes service account for the driver and attach the policy to the service account. Replacing the ARN of the policy with the ARN returned in the previous step.
+    ```
+    eksctl create iamserviceaccount \
+        --region $AWS_REGION \
+        --name fsx-csi-controller-sa \
+        --namespace kube-system \
+        --cluster $CLUSTER_NAME \
+        --attach-policy-arn arn:aws:iam::<111122223333:policy/Amazon_FSx_Lustre_CSI_Driver> \
```
Comment (brianhammons) [inline content/beginner/190_fsx_lustre/launching-fsx.md:107]: I think $ACCOUNT_ID should already be set if you want to add it.

File: content/beginner/190_fsx_lustre/launching-fsx.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+    clusterrolebinding.rbac.authorization.k8s.io/fsx-csi-external-provisioner-binding created
+    deployment.apps/fsx-csi-controller created
+    daemonset.apps/fsx-csi-node created
+    csidriver.storage.k8s.io/fsx.csi.aws.com created
+    ```
+
+6. Patch the driver deployment to add the service account that you created in step 3, replacing the ARN with the ARN that you noted in step 4.
+
+```
+kubectl annotate serviceaccount -n kube-system <fsx-csi-controller-sa> \
```
Comment (brianhammons) [inline content/beginner/190_fsx_lustre/launching-fsx.md:151]: should this be in brackets?

File: content/beginner/190_fsx_lustre/launching-fsx.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+ eks.amazonaws.com/role-arn=<arn:aws:iam::111122223333:role/eksctl-prod-addon-iamserviceaccount-kube-sys-Role1-NPFTLHJ5PJF5> --overwrite=true
```
Comment (brianhammons) [inline content/beginner/190_fsx_lustre/launching-fsx.md:152]: use $ROLE_ARN from step 4.

File: content/beginner/190_fsx_lustre/launching-fsx.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+    You'll see several lines of output as the service account is created. The last line of output is similar to the following example line.
+
+    ```
+    [ℹ]  created serviceaccount "kube-system/fsx-csi-controller-sa"
+    ```
+
+    Note the name of the AWS CloudFormation stack that was deployed. In the example output above, the stack is named eksctl-prod-addon-iamserviceaccount-kube-system-fsx-csi-controller-sa.
+
+4. Note the Role ARN for the role that was created.
```
Comment (brianhammons) [inline content/beginner/190_fsx_lustre/launching-fsx.md:121]: you can do step 4 from the cli:

export ROLE_ARN=$(aws cloudformation describe-stacks --stack-name eksctl-eksworkshop-eksctl-addon-iamserviceaccount-kube-system-fsx-csi-controller-sa --query "Stacks[0].Outputs[0].OutputValue")

**Review by brianhammons — CHANGES_REQUESTED** [review]
Can you review the changes and re-submit?

File: content/beginner/190_fsx_lustre/launching-fsx.md (hunk 5 above)
Comment (kornelbicskei) [reply content/beginner/190_fsx_lustre/launching-fsx.md:121]: hope the order of the output variables are the same hehe

File: content/beginner/190_fsx_lustre/cleaning.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+kubectl delete -k "github.com/kubernetes-sigs/aws-fsx-csi-driver/deploy/kubernetes/overlays/stable/?ref=master"
+
+aws s3 rm --recursive s3://$S3_LOGS_BUCKET
+aws s3 rb  s3://$S3_LOGS_BUCKET
+
+eksctl delete iamserviceaccount \
+    --region $AWS_REGION \
+    --name fsx-csi-controller-sa \
+    --namespace kube-system \
+    --cluster $CLUSTER_NAME \
```
Comment (brianhammons) [inline content/beginner/190_fsx_lustre/cleaning.md:24]: Extra \ at the end

File: content/beginner/190_fsx_lustre/cleaning.md [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+aws iam delete-policy \
+    --policy-name arn:aws:iam::$ACCOUNT_ID:policy/Amazon_FSx_Lustre_CSI_Driver
```
Comment (brianhammons) [inline content/beginner/190_fsx_lustre/cleaning.md:27]: --policy-arn

File: content/beginner/190_fsx_lustre/deploying-services.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+    kubectl exec -ti fsx-app -- lfs hsm_archive /data/out.txt
+    ```
+
+    New files aren't synced back to Amazon S3 automatically. In order to sync files to the s3ExportPath, you need to install the Lustre client in your container image and manually run the lfs hsm_archive command. The container should run in privileged mode with the CAP_SYS_ADMIN capability.
+    This example uses a lifecycle hook to install the Lustre client for demonstration purpose. A normal approach is building a container image with the Lustre client.
+
+3. Confirm that the out.txt file was written to the s3ExportPath folder in Amazon S3.
+
+    ```
+    aws s3 ls fsx-csi/export/
```
Comment (brianhammons) [inline content/beginner/190_fsx_lustre/deploying-services.md:141]: aws s3 ls s3://$S3_LOGS_BUCKET/export/

**Review by brianhammons — CHANGES_REQUESTED** [review]
Just some small changes to the cleanup and final s3 list call.

**Review by brianhammons — APPROVED** [review]
Looks good, thanks!

**Pull request merged**